### PR TITLE
Workshop parameter resolver

### PR DESCRIPTION
**Pull request by ghost**
You can find the description of the workshop in WORKSHOP.md. The classes with the solution in the solution-workshop folder.

### CODE REVIEW

File: solution-workshop/src/main/java/io/axoniq/demo/hotel/booking/query/RoomAvailabilityHandler.java [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+class RoomAvailabilityHandler {
+
+    private final RoomAvailabilityEntityRepository roomAvailabilityEntityRepository;
+    private final QueryUpdateEmitter queryUpdateEmitter;
+
+    private RoomAvailabilityResponseData convert(RoomAvailabilityEntity entity, UUID accountId) {
+        return new RoomAvailabilityResponseData(entity.getRoomNumber(),
+                                                entity.getRoomDescription(),
+                                                entity.getRoomStatus(),
+                                                accountId==null ? entity.getBookings().stream().map(it -> new BookingData(it.getId(), it.getStartDate(), it.getEndDate())).collect(Collectors.toUnmodifiableList()) : entity.getBookings().stream().filter(it -> it.getAccountId().equals(accountId.toString())).map(it -> new BookingData(it.getId(), it.getStartDate(), it.getEndDate())).collect(Collectors.toUnmodifiableList()),
```
Comment (Morlack) [inline solution-workshop/src/main/java/io/axoniq/demo/hotel/booking/query/RoomAvailabilityHandler.java:39]: This line is 422 characters wide, which is a bit much. Can we format this and use variables?

File: solution-workshop/src/main/java/io/axoniq/demo/hotel/booking/query/RoomAvailabilityEntityResolver.java [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+    private final RoomAvailabilityEntityRepository roomAvailabilityEntityRepository;
+
+    @Autowired
+    public RoomAvailabilityEntityResolver(RoomAvailabilityEntityRepository roomAvailabilityEntityRepository) {
+        this.roomAvailabilityEntityRepository = roomAvailabilityEntityRepository;
+    }
+
+    @Override
+    public RoomAvailabilityEntity resolveParameterValue(Message<?> message) {
+        if (matches(message)) {
```
Comment (Morlack) [inline solution-workshop/src/main/java/io/axoniq/demo/hotel/booking/query/RoomAvailabilityEntityResolver.java:26]: This `matches` call is not necessary, right? When this is invoked the matches call has already been invoked by the caller

File: WORKSHOP.md [hunk 3]
```diff
@@ -0,0 +1,74 @@
+# Parameter resolver workshop
+
+Welcome to the parameter resolver workshop
```
Comment (Morlack) [inline WORKSHOP.md:3]:
```suggestion
Welcome to the parameter resolver workshop!
```

File: WORKSHOP.md [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+
+In this workshop you will learn how to apply parameter resolvers.
+
+## Why should I use a parameter resolver?
+
+When using message handlers you can get into the situation that you need to do a lot of code duplication. For instance when you need to check if the user is authorized to apply a command. Or, event handlers that update a table need to get the entity in every function. You can do this check or fetch the entity in a parameter resolver and add it to the signature of any message handling function.
```
Comment (Morlack) [inline WORKSHOP.md:9]:
```suggestion
When using message handlers, you can get into a situation where you need to duplicate a lot of code. For instance, you need to check if the user is authorized to apply a command. Or, event handlers that update a table need to get the entity in every function. You can do this check or fetch the entity in a parameter resolver and add it to the signature of any message handling function.
```

File: WORKSHOP.md [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+
+A manager can add rooms, see the overall availability, check out, see payments and cleaning schedule. A guest can book a room, check in and pay.
+
+### Lab 1 Create a parameter resolver
+Run the HotelBooking application and use the [http request file](hotel-booking.http) or use the frontend to explore the application.
+
+Looking at the `RoomAvailabilityHandler`  you can see that the `RoomAvailabilityEntityRepository` is defined. Everytime this repository is updated the entity is fetched by calling `roomAvailabilityEntityRepository.getById(event.getRoomNumber())`. 
+The goal of this workshop is to create a parameter resolver that fetches this entity beforehand by using a parameter resolver.
+
+Add a component that `implements ParameterResolver<RoomAvailabilityEntity>, ParameterResolverFactory`. **(tip)** You need to Autowire the constructor of the component to make it being picked up by Spring.
```
Comment (Morlack) [inline WORKSHOP.md:50]:
```suggestion
Add a component that `implements ParameterResolver<RoomAvailabilityEntity>, ParameterResolverFactory`. By annotating it with `@Component` the configuration will automatically pick it up and register it.
```

**Review by CodeDrivenMitch — COMMENTED** [review]
Looks good in general, just needs some small changes

File: solution-workshop/src/main/java/io/axoniq/demo/hotel/booking/query/RoomAvailabilityHandler.java (hunk 1 above)
Comment (YvonneCeelie) [reply solution-workshop/src/main/java/io/axoniq/demo/hotel/booking/query/RoomAvailabilityHandler.java:39]: This is something we should fix in master

**Pull request closed without merging**